src/Ethereum.js

```diff
@@ -19,31 +19,33 @@ export default class StreamrEthereum {
         if (auth.privateKey) {
             const key = auth.privateKey
             const address = getAddress(computeAddress(key))
-            this._getAddress = () => address
+            this._getAddress = async () => address
             this.getSigner = () => new Wallet(key, this.getMainnetProvider())
             this.getSidechainSigner = async () => new Wallet(key, this.getSidechainProvider())
         } else if (auth.ethereum) {
-            this._getAddress = () => {
-                if (auth.ethereum.selectedAddress) {
+            this._getAddress = async () => {
+                try {
+                    const accounts = await auth.ethereum.request({ method: 'eth_requestAccounts' })
```

Comment:
Should probably cache this?

i.e. add `let address` in constructor

then
```js
this._getAddress = async () => {
            if (address) { return address }

            try {
                const accounts = await auth.ethereum.request({ method: 'eth_requestAccounts' })
                address = getAddress(accounts[0]) // convert to checksum case
                return address
            } catch {
                throw new Error('no addresses connected+selected in Metamask')
            }
        }
```

Though this doesn't prevent duplicate outstanding initial calls while `await auth.ethereum.request({ method: 'eth_requestAccounts' })` is pending, so perhaps wrap this in `pOne` or `CacheAsyncFn` from `src/utils/index.ts`

Reply:
Hm, though now I wonder what happens, and what should happen, if user does something like log out of metamask or switches metamask accounts after this stuff has been read by the client. Perhaps there's some "account changed" events or something we should be hooking into. Something to look into separately rather than in this ticket though.

Reply:
Yes, account changes are not implemented yet. 

If the client has already fetched the address from MetaMask, the subsequent call to client.getAddress() resolve automatically to the selected accounts without any UI interaction until clients disconnects that account. So in that sense no caching is needed.